src/ripple/shamap/impl/SHAMapTreeNode.cpp

```diff
@@ -377,6 +469,33 @@ SHAMapInnerNode::addRaw(Serializer& s, SHANodeFormat format) const
 }
 
 void
+SHAMapInnerNodeV2::addRaw(Serializer& s, SHANodeFormat format) const
+{
+    if (format == snfPREFIX)
+    {
+        s.add32 (HashPrefix::innerNodeV2);
+
+        for (int i = 0 ; i < 16; ++i)
+            s.add256 (mHashes[i].as_uint256());
+
+        s.add8(depth_);
+
+        auto x = common_.begin();
+        for (auto i = 0; i < (depth_+1)/2; ++i, ++x)
+            s.add8(*x);
+    }
+    else
+    {
+        SHAMapInnerNode::addRaw(s, format);
+        if (format == snfWIRE)
+        {
+            auto& data = s.modData();
+            data.back() += 3;
```

Comment:
Its a design flaw that the serialization algorithm for some of the objects is not an online algorithm (https://en.wikipedia.org/wiki/Online_algorithm). If possible we should take this opportunity to correct it.